src/Runtime/Distributions/Automata/Automaton.GroupExtractor.cs

```diff
@@ -25,7 +25,7 @@ private static class GroupExtractor
             {
                 Dictionary<int, HashSet<int>> subGraphs;
                 var order = ComputeTopologicalOrderAndGroupSubgraphs(automaton, out subGraphs);
-                return BuildSubautomata(automaton.states, order, subGraphs);
+                return BuildSubautomata(automaton.States.ToList(), order, subGraphs);
```

Comment:
I don't see why you need `List<State>` here. A `IReadOnlyList<State>` would work just fine in the code using it and so, you shouldn't need a `ToList()`.